tests/filecheck/parser-printer/region_name_clash.xdsl

```diff
@@ -0,0 +1,83 @@
+// RUN: xdsl-opt %s | xdsl-opt | filecheck %s
+
+// Check that SSA values and blocks can reuse names across regions
+
+module() {
+
+  // Two functions that share a basic block name and a value name
+  func.func() ["sym_name" = "id", "function_type" = !fun<[!i32], [!i32]>, "sym_visibility" = "private"] {
+    ^0(%0 : !i32):
+      func.return(%0 : !i32)
+  }
+
+  func.func() ["sym_name" = "id2", "function_type" = !fun<[!i64], [!i64]>, "sym_visibility" = "private"] {
+    ^0(%0 : !i64):
+      
+      func.return(%0 : !i64)
+  }
+
+
+  // Check that blocks in nested regions can clash names with the outer region blocks
+  func.func() ["sym_name" = "id2", "function_type" = !fun<[!i1], [!i1]>, "sym_visibility" = "private"] {
+    ^0(%0 : !i1):
+      scf.if(%0 : !i1) {
+        ^0:
+          scf.yield()
+      } {
+        ^0:
+          scf.yield()
+      }
+      func.return(%0 : !i1)
+  }
+
+  // Check that SSA names can be reused as long as they are defined after
+  func.func() ["sym_name" = "id2", "function_type" = !fun<[!i1], [!i1]>, "sym_visibility" = "private"] {
+    ^0(%0 : !i1):
+      scf.if(%0 : !i1) {
+        %1 : !i32 = arith.constant() ["value" = 3 : !i32]
+        scf.yield()
+      } {
+        scf.yield()
+      }
+      %1 : !i32 = arith.constant() ["value" = 3 : !i32]
+      func.return(%0 : !i1)
+  }
+
+}
+
+// CHECK: module() {
+
+// CHECK:   func.func() ["sym_name" = "id", "function_type" = !fun<[!i32], [!i32]>, "sym_visibility" = "private"] {
```

Comment:
Can you move these checks up to the functions they check? Also, I feel using `CHECK-NEXT` instead of `CHECK`might make sense.